ql/src/java/org/apache/hadoop/hive/ql/parse/TezCompiler.java

```diff
@@ -1566,13 +1569,38 @@ private void removeSemijoinsParallelToMapJoin(OptimizeTezProcContext procCtx)
 
       List<ExprNodeDesc> keyDesc = selectedMJOp.getConf().getKeys().get(posBigTable);
       ExprNodeColumnDesc keyCol = (ExprNodeColumnDesc) keyDesc.get(0);
-
-      tsProbeDecodeCtx = new TableScanOperator.ProbeDecodeContext(mjCacheKey, mjSmallTablePos,
-          keyCol.getColumn(), selectedMJOpRatio);
+      String realTSColName = getOriginalTSColName(selectedMJOp, keyCol.getColumn());
+      if (realTSColName != null) {
+        tsProbeDecodeCtx = new TableScanOperator.ProbeDecodeContext(mjCacheKey, mjSmallTablePos,
+                realTSColName, selectedMJOpRatio);
+      } else {
+        LOG.warn("ProbeDecode could not find TSColName for ColKey {} with MJ Schema {} ", keyCol, selectedMJOp.getSchema());
+      }
     }
     return tsProbeDecodeCtx;
   }
 
+  private static String getOriginalTSColName(MapJoinOperator mjOp, String internalCoName) {
```

Comment:
Can you check if this can already be done by any of the utility methods in `OperatorUtils`? If it is not, can we move it to that class with rest of utility methods in case it is useful in the future?

Reply:
Utility method is now moved to operatorUtils